underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileOutputStream.java

```diff
@@ -49,6 +49,8 @@ public void flush() throws IOException {
     //#ifdef HADOOP1
     mOut.sync();
     //#else
+    // Note that, hsync() flushes out the data in client's user buffer all the way to the disk
+    // device which may result in much slower journal performance than hsync().
```

Comment:
minor: I'd remove "journal" since the journal concept doesn't exist at this layer of abstraction